File: net/JetBrains.SignatureVerifier/Crypt/VerifySignatureResult.cs
namespace JetBrains.SignatureVerifier
{
    public class VerifySignatureResult
    {
        private readonly VerifySignatureStatus _status;
        public VerifySignatureStatus Status => _status;
        public bool NotValid => _status != VerifySignatureStatus.Valid;
        public string Message { get; set; }
        public static VerifySignatureResult Valid = new VerifySignatureResult(VerifySignatureStatus.Valid);
        public static VerifySignatureResult NotSigned = new VerifySignatureResult(VerifySignatureStatus.NotSigned);

        public VerifySignatureResult(VerifySignatureStatus status)
        {
            _status = status;
        }

        public static VerifySignatureResult InvalidChain(string message) =>
            new VerifySignatureResult(VerifySignatureStatus.InvalidChain)
                { Message = message };
    }

    public enum VerifySignatureStatus
    {
        Valid,
        NotSigned,
        InvalidSignature,
        InvalidChain,
        InvalidTimestamp
    }
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/crypt/VerifySignatureResult.kt
package com.jetbrains.signatureverifier.crypt

import org.bouncycastle.cert.X509CertificateHolder
import java.security.cert.X509Certificate

class VerifySignatureResult(
  private val status: VerifySignatureStatus,
  private val message: String? = null,
  certificate: X509CertificateHolder? =null
) {
  val Status: VerifySignatureStatus
    get() = status

  val Message: String?
    get() = message

  val NotValid: Boolean
    get() = status != VerifySignatureStatus.Valid

  companion object {
    val Valid = VerifySignatureResult(VerifySignatureStatus.Valid, certificate = null)

    fun InvalidChain(message: String): VerifySignatureResult =
      VerifySignatureResult(VerifySignatureStatus.InvalidChain, message, null)
  }
}

enum class VerifySignatureStatus {
  Valid,
  InvalidSignature,
  InvalidChain,
  InvalidTimestamp
}
